internal/pkg/runtime/launcher/oci/process_linux.go

```diff
@@ -286,6 +289,26 @@ func getProcessEnv(imageSpec imgspecv1.Image, runtimeEnv map[string]string) []st
 		}
 	}
 
+	// Handle PATH differently beteween OCI and native images
+	if nativeSIF {
+		setNativePath(g, prependPath, path, appendPath)
+	} else {
+		setOCIPath(g, prependPath, path, appendPath)
+	}
+
+	// Ensure LD_LIBRARY_PATH always contains singularity lib binding dir.
+	// This is handled by environment scripts in native SIF images.
+	if !nativeSIF && !strings.Contains(ldLibraryPath, singularityLibs) {
+		ldLibraryPath = strings.TrimPrefix(ldLibraryPath+":"+singularityLibs, ":")
+	}
+	if ldLibraryPath != "" {
+		g.AddProcessEnv("LD_LIBRARY_PATH", ldLibraryPath)
+	}
+
+	return g.Config.Process.Env
+}
+
+func setOCIPath(g *generate.Generator, prependPath, path, appendPath string) {
 	// Compute and set optionally APPEND-ed / PREPEND-ed PATH.
 	if appendPath != "" {
 		path = path + ":" + appendPath
```

Comment:
Just making sure: we don't want to do the same `strings.TrimPrefix(..., ":")` here that we do in line 302 of the same file? After all, if `path` being empty is something we're bothering checking for, it stands to reason (I think...?) that `path` can be empty also in a circumstance where `appendPath` is not empty (and therefore the concatenation will occur).

Reply:
Yeah, could do that for neatness and consistency. It doesn't have any practical effect, and PATH is very unlikely to be "" but I'll make the change.